airflow/providers/snowflake/operators/snowflake_to_slack.py

```diff
@@ -0,0 +1,149 @@
+    results_df }}'. The 'results_df' variable name can be changed by specifing a different
+    'results_df_name' parameter. The Tabulate library is added to the JINJA environment as a filter to
+    allow the dataframe to be rendered nicely. For example, set 'slack_message' to {{ results_df |
+    tabulate(tablefmt="pretty", headers="keys") }} to send the results to Slack as an ascii rendered table.
+
+
+    :param snowflake_conn_id: The Snowflake connection id
+    :type snowflake_conn_id: str
+    :param sql: The SQL statement to execute on Snowflake (templated)
+    :type sql: str
+    :param results_df_name: The name of the JINJA template's dataframe variable, default is 'results_df'
+    :type results_df_name: str
+    :param parameters: The parameters to pass to the SQL query
+    :type parameters: mapping or iterable
+    :param warehouse: The Snowflake virtual warehouse to use to run the SQL query
+    :type warehouse: str
+    :param database: The Snowflake database to use for the SQL query
+    :type database: str
+    :param schema: The schema to run the SQL against in Snowflake
+    :type schema: str
+    :param role: The role to use when connecting to Snowflake
+    :type role: str
+    :param slack_conn_id: The connection id for Slack
+    :type slack_conn_id: str
+    :param slack_token: The token to use to authenticate to Slack. If this is not provided, the
+        'webhook_token' attribute needs to be specified in the 'Extra' JSON field against the slack_conn_id
+    :type slack_token: str
+    :param slack_message: The templated Slack message to send with the data returned from Snowflake.
+        You can use the default JINJA variable {{ results_df }} to access the pandas dataframe containing the
+        SQL results
+    :type slack_message: str
+    """
+    template_fields = ('sql', 'slack_message')
+    template_ext = ('.sql', '.jinja', '.j2',)
+    times_rendered = 0
+
+    # pylint: disable=too-many-arguments
+    @apply_defaults
+    def __init__(self,
+                 sql,
+                 slack_message,
+                 snowflake_conn_id='snowflake_default',
+                 slack_conn_id='slack_default',
+                 results_df_name='results_df',
+                 parameters=None,
+                 warehouse=None,
+                 database=None,
+                 schema=None,
+                 role=None,
+                 slack_token=None,
+                 *args, **kwargs):
+        super(SnowflakeToSlackOperator, self).__init__(*args, **kwargs)
+
+        self.snowflake_conn_id = snowflake_conn_id
+        self.sql = sql
+        self.parameters = parameters
+        self.warehouse = warehouse
+        self.database = database
+        self.schema = schema
+        self.role = role
+        self.slack_conn_id = slack_conn_id
+        self.slack_token = slack_token
+        self.slack_message = slack_message
+        self.results_df_name = results_df_name
+
+    def _get_query_results(self):
```

Comment:
Would you mind adding type hints to functions? It makes code easier to understand :)